server/lib/flow-metrics.js

```diff
@@ -4,29 +4,56 @@
 
 var crypto = require('crypto');
 
-/**
- *
- * @param key String flow_id_key
- * @param userAgent String request user agent string
- * @returns {{flowBeginTime: number, flowId: string}}
- */
-module.exports = function flowEventData(key, userAgent) {
-  var flowId = crypto.randomBytes(16).toString('hex');
-  var flowBeginTime = Date.now();
+const SUBSTRING_LENGTH = 32;
 
+module.exports = {
+  /**
+   * Create flow data.
+   *
+   * @param key String
+   * @param userAgent String
+   * @returns {{flowBeginTime: number, flowId: string}}
+   */
+  create (key, userAgent) {
+    const salt = crypto.randomBytes(16).toString('hex');
+    return createflowEventData(key, salt, Date.now(), userAgent);
+  },
+
+  /**
+   * Check whether a flowId is valid.
+   *
+   * @param flowBeginTime Number
+   * @param key String
+   * @param userAgent String
+   * @returns Boolean
+   */
+  check (flowId, key, flowBeginTime, userAgent) {
```

Comment:
tiny, tiny api nit: the ordering of the arguments to this function seems arbitrary; putting `key` first might make for more pleasing symmetry with the API of the `create` function.